demos/common/cpp/models/src/model_base.cpp

```diff
@@ -54,9 +55,18 @@ InferenceEngine::CNNNetwork ModelBase::prepareNetwork(InferenceEngine::Core& cor
 
 ExecutableNetwork ModelBase::loadExecutableNetwork(const CnnConfig& cnnConfig, InferenceEngine::Core& core) {
     this->cnnConfig = cnnConfig;
-    auto cnnNetwork = prepareNetwork(core);
+    // ---------------------- Determine network type ---------------------------------------------------
+    isNetworkCompiled = fileExt(modelFileName) == "blob";
```

Comment:
You should determine model "type" as soon as you get filename - in `ModelBase::prepareNetwork` method or even in constructor. It also will prevent reading network.